subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/r36/BuildProgressCrossVersionSpec.groovy

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r36
+
+import org.gradle.integtests.tooling.fixture.ProgressEvents
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.test.fixtures.file.LeaksFileHandles
+import org.gradle.test.fixtures.maven.MavenFileRepository
+import org.gradle.test.fixtures.server.http.MavenHttpRepository
+import org.gradle.test.fixtures.server.http.RepositoryHttpServer
+import org.gradle.tooling.ProjectConnection
+import org.junit.Rule
+
+@ToolingApiVersion(">=3.6")
+@TargetGradleVersion(">=3.6")
+class BuildProgressCrossVersionSpec extends ToolingApiSpecification {
```

Comment:
Given the awkward wiring, it might be good to add some basic coverage that we get some events for kotlin build scripts.